src/main/java/com/nirima/jenkins/plugins/docker/DockerCloud.java

```diff
@@ -361,26 +365,32 @@ int countContainersInProgress() {
                         t.getImage(), label, getDisplayName());
 
                 final CompletableFuture<Node> plannedNode = new CompletableFuture<>();
-                r.add(new NodeProvisioner.PlannedNode(t.getDisplayName(), plannedNode, t.getNumExecutors()));
+                DockerNodeFactory.DockerPlannedNode pn =
+                        NODE_FACTORY.createPlannedNode(t.getDisplayName(), plannedNode, t.getNumExecutors(),
+                                getDisplayName(), t.getImage(), label.getName());
 
+                r.add(pn);
                 final Runnable taskToCreateNewSlave = new Runnable() {
                     @Override
                     public void run() {
                         DockerTransientNode slave = null;
                         try {
-                            // TODO where can we log provisioning progress ?
+                            // TODO where can we log provisioning progress ? // In the CloudMonitor API!
```

Comment:
If you've resolved the TODO comment, please remove the TODO comment and replace it with a comment explaining where we log provisioning progress.
i.e. once it's done, it's no longer TODO :wink: